File: exercises/beecrowd/1151/1151.ts
// https://judge.beecrowd.com/pt/problems/view/1151
import * as fs from "fs";

function main(): void {
  const n = parseInt(fs.readFileSync("/dev/stdin", "utf8").trim());

  let a = 0,
    b = 1;
  let result = "";

  for (let i = 0; i < n; i++) {
    result += i === 0 ? `${a}` : ` ${a}`;
    const next = a + b;
    a = b;
    b = next;
  }

  console.log(result);
}

main();

File: exercises/beecrowd/1151/1151.go
// https://judge.beecrowd.com/pt/problems/view/1151
package main

import "fmt"

func main() {
	var n int
	fmt.Scan(&n)

	a, b := 0, 1

	for i := 0; i < n; i++ {
		if i == 0 {
			fmt.Print(a)
		} else {
			fmt.Print(" ", a)
		}
		a, b = b, a+b
	}
	fmt.Println()
}
